### ISSUE TITLE
ES2303-af57ba9e - Insufficient protection of 2FA SMS Code

### ISSUE THREAD

**Issue by cmullaly-mitre** [CWE-CAPEC/CWE-Content-Development-Repository#55]
Submission File: [ES2303-af57ba9e-insufficient-protection-of-2fa-sms-code.txt](https://github.com/CWE-CAPEC/CWE-Content-Development-Repository/blob/main/submissions/ES2303-af57ba9e-insufficient-protection-of-2fa-sms-code.txt)

ID: ES2303-af57ba9e

SUBMISSION DATE: 2023-03-31 16:32:59

NAME: Insufficient protection of 2FA SMS Code

DESCRIPTION:

When 2FA SMS code is displayed at the beginning of the SMS message, quite
often it is possible to get to know code value without unblock device
action. That means, bad actor without knowing how to unblock victim device
can get to know this sensitive information, because it is displayed often
in 'bar' - shortcut/preview of message. Safer approach is to place this
code after part displayed message's shortcut. Example: victim leave a
mobile device in the room, because it is PIN or pattern protected and go
out for few minutes, bad actor from the same room can try exploit. Example:
'CODE Lorem ips--um' vs 'Lorem ipsum C--ODE' ; here "--" means ending of
visible part without unblocked device. Thanks to this real user who know
how to unblock device will get to know 2FA SMS code, but bad actor will can
not exploit this. This scenario assumes that bad actor is close to victim's
device - physicial part, but message form with 2FA Code is software part.

**Comment by stevechristeycoley** [comment]
This submission has completed the INITIAL REVIEW phase by a CWE Content Team member, who reviewed the submission based on criteria that are intended to ensure that new CWE content meets minimum quality expectations and falls within the current scope of CWE, which simplifies their integration into CWE. (As of January 2024, these criteria are still under development).

- [More details on submission problems](https://github.com/CWE-CAPEC/CWE-Content-Development-Repository/blob/main/documentation/submission-problems.md)
- [More details on scope exclusions](https://github.com/CWE-CAPEC/CWE-Content-Development-Repository/blob/main/documentation/scope-exclusions.md)

Please use ID ES2303-af57ba9e in future correspondence regarding this request and copy cwe-submissions@mitre.org on any inquiry or reply.

This submission contains one or more issues that will require some changes before the submission can be accepted, so it has been moved to the Initial Consultation phase.

The Initial Review has found the following issues, which are summarized. "Resolved" issues (if any) were considered during review but were already addressed, so they require no action.  Detailed explanations are included in the full text of your submission; see "ACTIVE ISSUES - DETAILS" and "RESOLVED ISSUES - DETAILS" sections.

ACTIVE ISSUES:

* SUB.ABS.LOWLEVEL "Low level of abstraction"
* SUB.RELREFS "No relevant references"
* SUB.ACTION "Unable to determine action"


RESOLVED ISSUES:

None.



Underneath the boundary below, please read the "Comments:" for each issue, enter your "Response:" to each comment, and suggest changes to the submission text. CWE Team members will make the appropriate updates to the raw submission file.

=================================================================


## SUB.ABS.LOWLEVEL - "Low level of abstraction"

URL: [SUB.ABS.LOWLEVEL](https://github.com/CWE-CAPEC/CWE-Content-Development-Repository/blob/main/documentation/submission-problems.md#SUB.ABS.LOWLEVEL)

*Description*: The submission identifies an issue, but its abstraction might
be too low-level to merit its own CWE entry. If it is focused on a
particular technology, framework, protocol, or function, then it might not
be unique enough to get its own CWE ID.  This might lead to modification of
an existing entry to include the additional details, or creation of a
higher-level entry for which the submission is one example. (Note: some
older CWE entries have this problem and would not be published today,
generally because they were too specific to a particular language.)

*Resolution*: The submission will be allowed to progress to later phases once
the more-general concept is appropriately identified and described. The
original, lower-level submission will serve as a primary example of the
more general concept.

*Comments*: While the submission title broadly addresses the lack of 
protection for SMS codes, the description focuses narrowly on the issue 
of SMS codes being visible through message previews. This specific 
concern seems more relevant to end-users than developers, as users 
can choose to disable message previews on their devices to mitigate the risk.<br><br>From a developer's perspective, addressing this issue is challenging due 
to the inherent limitations of SMS technology, which treats all messages 
as plain text. In many cases, developers may not even have the option to 
switch the second factor from SMS to a more secure method, as SMS-based 
authentication might be the default or only available option provided 
by the application.<br><br>Given the narrow scope of the issueâ€”limited to SMSâ€”and the broader 
complexities surrounding it, this concern appears too low-level to 
warrant a new entry.

*Response*: \<SUBMITTER SHOULD REPLY HERE\><br><br>




## SUB.RELREFS - "No relevant references"

URL: [SUB.RELREFS](https://github.com/CWE-CAPEC/CWE-Content-Development-Repository/blob/main/documentation/submission-problems.md#SUB.RELREFS)

*Description*: The submission does not provide relevant references that
explain the weakness, or the relevance is not easily found.

*Resolution*: The submission is not likely to progress to later phases if
there are no applicable, easily-accessible references. The submitter could
resolve by providing additional details such as a section name and quote
from the reference.  If no suitable references can be found, the CWE Team
may decide to make an exception.

*Comments*: The submission doesn't include any actual references. Please
provide valid reference(s) to justify the real-world example of the issue.

*Response*: \<SUBMITTER SHOULD REPLY HERE\><br><br>




## SUB.ACTION - "Unable to determine action"

URL: [SUB.ACTION](https://github.com/CWE-CAPEC/CWE-Content-Development-Repository/blob/main/documentation/submission-problems.md#SUB.ACTION)

*Description*: It is not clear what action to perform based on the
submission, e.g., create new entry, update existing entry, etc.

*Resolution*: The submission cannot progress to the Internal-Update stage
until the relevant actions(s) are identified.

*Comments*: Based on the other submission issues identified above, the CWE
team isn't clear on the next steps. We likely won't be able to accept
this submission. 

*Response*: \<SUBMITTER SHOULD REPLY HERE\>

**Comment by stevechristeycoley** [comment]
This submission has been accepted (Phase 6) and will move to a new stage. This means that the CWE Team has agreed to make the suggested change(s) to the relevant CWE entry/entries and include this change in an upcoming version.

- If we need to obtain additional details or work closely on finalizing changes before modifying the CWE entry/entries, we will move to Stage 2, Phase 7: Det-Sub-Requested.

- If no further changes are needed, then we will move to Stage 3, Phase 12: Internal-Update, where we will begin to modify CWE content directly.

You should receive notification about the next phase change shortly.

## CREDITS/ACKNOWLEDGMENTS FOR YOUR CONTRIBUTION:

We plan to credit you using your original contact and organization information. See your email notification for specifics.

* If you wish to change this information, please let us know.

* If you wish to remain anonymous, we will credit an "Anonymous External
  Contributor".

* If we do not receive a response from you, then we will
  use your GitHub username (if any).

Any issues listed below are still active but will be addressed in a
later phase. Generally, no response is necessary at this time.

Thank you for supporting CWE!

The CWE Team

=================================================================

## SUB.ACTCHANGE - "Change in action for submission"

URL: [SUB.ACTCHANGE](https://github.com/CWE-CAPEC/CWE-Content-Development-Repository/blob/main/documentation/submission-problems.md#SUB.ACTCHANGE)

*Description*: The submission's action needs to change, e.g., from "New
Entry" to "Modification," or vice versa. This potentially requires changes
to the kinds of information that will need to be provided.

*Resolution*: The submission cannot progress to Stage 3 (Content-Generation)
until the informational needs are addressed.

*Comments*: While there is a good argument for creating a new CWE entry,
it would overlap other CWE entries in the current hierarchical
organization, and we will not be able to make such a large adjustment
at this time. We will address this by modifying both CWE-201
(Insertion of Sensitive Information Into Sent Data) - which is a
ChildOf CWE-200 - as well as CWE-359: Exposure of Private Personal
Information to an Unauthorized Actor. Modifications will include (1)
writing a detailed demonstrative example, using your submission as a
basis; (2) citing some references we found, which we've added to this
submission; (3) defining "SMS preview exposure" as an alternate term;
(4) providing some mitigation, using information from your submission
and the references; and (5) including Mobile within the
Applicable_Platforms for the two entries.

*Response*: [NO RESPONSE NECESSARY; SUBMISSION HAS BEEN ACCEPTED AS-IS.]
